rumqttc/src/v5/eventloop.rs

```diff
@@ -168,7 +168,19 @@ impl EventLoop {
             Err(e) => {
                 // MQTT requires that packets pending acknowledgement should be republished on session resume.
                 // Move pending messages from state to eventloop.
-                self.clean();
+                // The PubAckFail/PubRecFail/PubRelFail/PubCompFail/SubFail/UnsubFail errors are not critical errors.
+                match e {
+                    ConnectionError::MqttState(StateError::PubAckFail { .. })
+                    | ConnectionError::MqttState(StateError::PubRecFail { .. })
+                    | ConnectionError::MqttState(StateError::PubRelFail { .. })
+                    | ConnectionError::MqttState(StateError::PubCompFail { .. })
+                    | ConnectionError::MqttState(StateError::SubFail { .. })
+                    | ConnectionError::MqttState(StateError::UnsubFail { .. }) => {}
```

Comment:
instead of matching them here and ignoring, we can just remove those variants from `StateError` right?

i think that will be much better. wdyt?

Reply:
Sounds not bad.
Considering the acknowledgement is finished then, `handle_incoming_...ack` could return `Ok(None)`, since there should not be any outgoing packet. `Err(...)` is not used because there is no state error.

Reply:
other logic will remain same ( as there can be collisions and unsolicited / OOO acks ). Only the part where we check for status code ( and return error lets say `PubAckFail` ) will be removed.

Reply:
Updated code.
Keep some branch for warning logs.